HuntTheWumpus/HuntTheWumpus.cpp

```diff
@@ -201,66 +200,90 @@ int main()
 
     while (change.IsPlaying())
     {
+
         // Parse input.
         std::string input;
 
         std::cout << "Command? " << std::flush;
 
-        std::getline(std::cin, input);
-
-        // Split into strings.
-        const auto stringTokens = SplitString(input, " \t\n");
-
-        const auto command = ToLower(stringTokens[0]);
-
-        if (command[0] == 'm')
+        try
         {
-            if (stringTokens.size() < 2)
+            std::getline(std::cin, input);
+            // Split into strings.
+            const auto stringTokens = SplitString(input, " \t\n");
+
+            if (stringTokens.size() == 0)
             {
-                std::cout << "A Move command must be followed by the destination cave id.\n";
-                continue;
+                throw std::runtime_error("");
             }
 
-            // Second token is a destination.
-            const auto destCave = std::stoi(stringTokens[1]);
-
-            dungeon.MakeMove(HuntTheWumpus::DungeonMove::Move, { destCave });
-        }
+            const auto command = ToLower(stringTokens[0]);
 
-        if (command[0] == 's')
-        {
-            if (stringTokens.size() < 2)
+            if (command[0] == 'm' && command.length() != 0)
             {
-                std::cout << "A Shoot command must be followed by a list of caves for the arrow to go through.\n";
-                continue;
-            }
+                if (stringTokens.size() < 2)
+                {
+                    std::cout << "A Move command must be followed by the destination cave id.\n";
+                    continue;
+                }
+
 
-            // Remaining tokens is the desired arrow path.
-            std::vector<int> path;
-            auto firstToken = false;
+                // Second token is a destination.
+                const auto destCave = std::stoi(stringTokens[1]);
```

Comment:
Can this fail in a way that's not addressed or be confused with a different recovery?